aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -0,0 +1,340 @@
+        }
+
+        /**
+         * A handler that will be invoked whenever there is a conflict between two model instances,
+         * one in the local store, and one from the remote server, as received from a sync operation.
+         * @param dataStoreConflictHandler A handler to invoke upon sync conflicts
+         * @return Current builder
+         */
+        @NonNull
+        public Builder dataStoreConflictHandler(@NonNull DataStoreConflictHandler dataStoreConflictHandler) {
+            this.dataStoreConflictHandler = Objects.requireNonNull(dataStoreConflictHandler);
+            return Builder.this;
+        }
+
+        /**
+         * Sets a handler function to be applied when the DataStore encounters an unrecoverable error
+         * in one of its ongoing background operations (model synchronization).
+         * @param dataStoreErrorHandler A handler for unrecoverable background errors
+         * @return Current builder instance
+         */
+        @NonNull
+        public Builder dataStoreErrorHandler(@NonNull DataStoreErrorHandler dataStoreErrorHandler) {
+            this.dataStoreErrorHandler = Objects.requireNonNull(dataStoreErrorHandler);
+            return Builder.this;
+        }
+
+        /**
+         * Sets the duration of time after which delta syncs will not be preferred over base syncs.
+         * @param syncIntervalMs The amount of time that must elapse for delta syncs to not be considered
+         * @return Current builder instance
+         */
+        @NonNull
+        public Builder syncIntervalMs(@IntRange(from = 0) long syncIntervalMs) {
+            //Only set this value if the incoming value is not equal to the default
+            if (syncIntervalMs != DEFAULT_SYNC_INTERVAL_MINUTES) {
+                this.syncIntervalMs = syncIntervalMs;
+            }
+            return Builder.this;
+        }
+
+        /**
+         * Sets the maximum number of records, from the server, to process from a sync operation.
+         * @param syncMaxRecords Max number of records client will consumer from server at one time
+         * @return Current builder instance
+         */
+        @NonNull
+        public Builder syncMaxRecords(@IntRange(from = 0) int syncMaxRecords) {
+            //Only set this value if the incoming value is not equal to the default
+            if (syncMaxRecords != DEFAULT_SYNC_MAX_RECORDS) {
+                this.syncMaxRecords = syncMaxRecords;
+            }
+            return Builder.this;
+        }
+
+        /**
+         * Sets the number of items requested in each page of sync results.
+         * @param syncPageSize Number of items requested per page in sync operation
+         * @return Current builder
+         */
+        @NonNull
+        public Builder syncPageSize(@IntRange(from = 0) int syncPageSize) {
+            //Only set this value if the incoming value is not equal to the default
+            if (syncPageSize != DEFAULT_SYNC_PAGE_SIZE) {
```

Comment:
Shouldn't need this `if`. In the case that `syncPageSize == DEFAULT_SYNC_PAGE_SIZE`, you'd just set the value to the same thing again. Which wouldn't change the system behavior.